templates/shared_services/nexus-cert/terraform/staticweb.tf

```diff
@@ -0,0 +1,30 @@
+data "azurerm_client_config" "deployer" {}
+
+resource "azurerm_storage_account" "staticweb" {
```

Comment:
Do we have anything in the repo that explains why we are deploying the storage static site, and how it is used?

Reply:
I've not added anything into the documentation about how it is used for this use case.  Although its very similar to how the cert is acquired for the API using letsencrypt.  Looking at the docs, the only reference I can find to any of this is references to letsencrypt in `deploying-azure-tre.md`.

If this is required to be documented for both letsencrypt use cases, I can raise an additional issue for this to prevent any delays in resolving #1089 by doing the documentation as part of this PR.

Reply:
Good point, I think that there are two potential cases of documentation here:

1. User docs. Is there anything that the user needs to do to trigger the lets-encrypt cert process for nexus? If so, that should be documented in user-facing docs
2. Developer/code docs. I think that the process we're using here is quite clever. But "clever" doesn't mean that it's obvious, so I was suggesting that some docs/notes are added to the code

In my initial comment, I was referring to the code docs, which I think would be good to include as part of the code in this PR so that future developers working on this code can quickly get an overview of how the letsencrypt process works and what the various components are used for. (potentially including a diagram similar to [this](https://github.com/microsoft/AzureTRE/issues/1089#issuecomment-1063928777)?)

Reply:
I'll get something together relating to the letsencrypt and how it sits together/is used this afternoon for both the additions in this PR and the existing structure for the other Letsencrypt use case.

As for the user docs I imagine 1/2 would be related to changes in this PR, 1/2 to the changes in #1479.  

Ideally I imagine a single `make` command would be ideal to deploy the resources required to get the cert, generate the cert and deploying nexus (replacing the existing `make nexus-install`? which is currently run as part of `make deploy-shared-services`, with each part still being able to be run independently.